src/app/shared/user.service.ts

```diff
@@ -1,55 +1,36 @@
 import { Injectable } from '@angular/core';
-import { FormBuilder, Validators, FormGroup } from '@angular/forms';
-import { HttpClient, HttpHeaders } from "@angular/common/http";
+import { HttpClient } from "@angular/common/http";
+import { UserComponent } from 'src/app/user/user.component'
 @Injectable({
   providedIn: 'root'
 })
 export class UserService {
 
-  constructor(private fb: FormBuilder, private http: HttpClient) { }
-  private readonly BaseUri = 'https://localhost:44377/api';
-  formModel = this.fb.group({
-    UserName : ['',Validators.required],
-    Email : ['',Validators.email],
-    FullName : [''],
-    Passwords : this.fb.group({
-      Password : ['',[Validators.required,Validators.minLength(3)]],
-      ConfirmPassword : ['',Validators.required]
-    },{validator : this.comparePasswords})
-  });
-
-  comparePasswords(fb:FormGroup) {
-    let confirmPswrdCtrl = fb.get('ConfirmPassword');
-    //passwordMismatch
-    //confirmPswrdCtrl.errors={passwordMismatch=true}
-    if(confirmPswrdCtrl.errors == null || 'passwordMismatch' in confirmPswrdCtrl.errors) {
-      if(fb.get('Password').value!=confirmPswrdCtrl.value) {
-        confirmPswrdCtrl.setErrors({ passwordMismatch:true});
-      }
-      else {
-        confirmPswrdCtrl.setErrors(null);
-      }
-    }
-  }
+  constructor(private http: HttpClient, private user : UserComponent) { }
+  private readonly BaseUri = 'https://localhost:44335/api';
 
   register() {
     var body = {
-      UserName : this.formModel.value.UserName,
-      Email : this.formModel.value.Email,
-      FullName : this.formModel.value.FullName,
-      Password : this.formModel.value.Passwords.Password,
+      Name : this.user.formModel.value.Name,
```

Comment:
Again, why this user component is defined in the User Service?
Please declare all these objects in component and send that object to the service. 

Service does not have to object creation logic

Reply:
done